R/OptimInstance.R

```diff
@@ -120,6 +120,41 @@ OptimInstance = R6Class("OptimInstance",
       assert_names(names(y), permutation.of = self$objective$codomain$ids())
       x_domain = transform_xdt_to_xss(xdt, self$search_space)[[1]]
       private$.result = cbind(xdt, x_domain = list(x_domain), t(y)) # t(y) so the name of y stays
+    },
+
+    #' @description
+    #' Evaluates (untransformed) points of only numeric values, and returns a
+    #' scalar objective value, where the return value is negated if the measure
+    #' is maximized. Internally, `$eval_batch()` is called with a single row.
+    #' This function serves as a objective function for optimizers of numeric
+    #' spaces - which should always be minimized.
+    #'
+    #' @param x (`numeric()`)\cr
+    #' Untransformed points.
+    #'
+    #' @return Objective value as `numeric(1)`.
+    objective_function = function(x) {
+      assert_numeric(x, len = self$search_space$length)
+      xs = set_names(as.list(x), self$search_space$ids())
+      self$search_space$assert(xs)
+      xdt = as.data.table(xs)
+      res = self$eval_batch(xdt)
+      y = as.numeric(res[, self$objective$codomain$ids()[1], with=FALSE])
+      if(self$objective$codomain$tags[[1]] == "minimize") y else -y
+    },
+
+    #' @description
+    #' Returns lower bounds of search space.
+    #' @return `numeric(1)`
+    objective_lower = function() {
```

Comment:
I think this is not a great name. It sound like it would return the lower bound of the codomain. I would just leave `objective_lower` and `objective_upper` out, or is there a specific need?

Reply:
We wanted to provide the most important arguments of `stats::optimize()`. However, the user could just use `inst$search_space$lower` for this

Reply:
> inst$search_space$lower seems enough.
dont redundantly replicate this - if there is not immediate need.
this just makes thinks confusing and boats code. search_space$lower is super clear!

IF OTOH the user has to do a 5 line "data-transformation" to access the results he wants, OR the access is
like with 50 "dollar-signs" down a complicated hierarchy such an AB is in order.
But our philosophy is: do it conservatively, not prematurely, when need is obvoius